File: application/src/test/java/com/smtm/application/users/endpoint/v1/FakeUserRegistration.java
package com.smtm.application.users.endpoint.v1;

import org.jetbrains.annotations.NotNull;
import com.smtm.users.api.UserRegistration;
import com.smtm.users.registration.UnsecuredPassword;
import com.smtm.users.registration.UserProfile;
import com.smtm.users.registration.UserProfileKt;

public class FakeUserRegistration implements UserRegistration {

    private UserProfile userProfile = UserProfileKt.validUserProfileOf(0, "noname@example.com");

    @NotNull
    @Override
    public UserProfile register(@NotNull String email, @NotNull UnsecuredPassword password) {
        return userProfile;
    }

    void returns(UserProfile userProfile) {
        this.userProfile = userProfile;
    }
}

File: application/src/test/kotlin/com/smtm/application/security/FakeUserRegistration.kt
package com.smtm.application.security

import com.smtm.security.api.UserRegistration
import com.smtm.security.registration.EmailAddress
import com.smtm.security.registration.UnencryptedPassword
import com.smtm.security.registration.UserProfile
import com.smtm.security.registration.validUserProfileOf

class FakeUserRegistration : UserRegistration {

    var userProfile: UserProfile? = null

    override fun register(email: EmailAddress, password: UnencryptedPassword): UserProfile = userProfile ?: validUserProfileOf(1, email)
}
